### PR TITLE
[19.0][MIG] sale_resource_booking: Migration to 19.0

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT64004

ping @pilarvargas-tecnativa @carlos-lopez-tecnativa

### CODE REVIEW

File: sale_resource_booking/wizards/resource_booking_sale.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    def action_generate(self):
+        order = self.env["sale.order"].create(
+            {
+                "partner_id": self.partner_id.id,
+                "order_line": [
+                    (
+                        0,
+                        0,
+                        {
```
Comment (carlos-lopez-tecnativa) [inline sale_resource_booking/wizards/resource_booking_sale.py]: Could you use the new Command syntax instead of the older syntax?
```suggestion
                    Command.create(
                        {
```

File: sale_resource_booking/wizards/resource_booking_sale.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            }
+        )
+        return {
+            "res_id": order.id,
+            "res_model": "sale.order",
+            "target": "current",
+            "type": "ir.actions.act_window",
+            "view_mode": "form",
+            "views": [[False, "form"]],
+        }
```
Comment (carlos-lopez-tecnativa) [inline sale_resource_booking/wizards/resource_booking_sale.py]: This can be simplified by using the [_get_records_action](https://github.com/odoo/odoo/blob/c2a39085ba0fbcf8a0e6a55228191e764499caea/odoo/addons/base/models/ir_ui_view.py#L3248) function


```suggestion
        return order._get_records_action()
```

File: sale_resource_booking/tests/test_sale_resource_booking.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        order_f = Form(self.env["sale.order"])
+        order_f.partner_id = self.partner
+        with order_f.order_line.new() as line_f:
+            line_f.product_id = self.product
+        order = order_f.save()
+        order.action_confirm()
+        bookings = order.order_line.mapped("resource_booking_ids")
+        # Open wizard
+        wizard = self.env["sale.order.booking.confirm"].create(
+            {"order_id": order.id, "resource_booking_ids": [(6, 0, bookings.ids)]}
```
Comment (carlos-lopez-tecnativa) [inline sale_resource_booking/tests/test_sale_resource_booking.py]:
```suggestion
            {"order_id": order.id, "resource_booking_ids": [Command.set(bookings.ids)]}
```

**Review by carlos-lopez-tecnativa — CHANGES_REQUESTED** [review]
Additionally, could you add some demo data to make it easier to test the module, such as a product configured with the minimum required fields?

**Comment by CarlosRoca13** [comment]
Done @carlos-lopez-tecnativa

**Review by carlos-lopez-tecnativa — APPROVED** [review]
LGTM

**Review by pilarvargas-tecnativa — APPROVED** [review]
